featuretools/primitives/standard/transform_primitive.py

```diff
@@ -748,3 +751,127 @@ def url_to_protocol(x):
             p = r'^(https|http)(?:\:)'
             return x.str.extract(p, expand=False)
         return url_to_protocol
+
+
+class URLToTLD(TransformPrimitive):
+    """Determines the top level domain of a url.
+    Description:
+        Extract the top level domain of a url, using regex,
+        and a list of common top level domains. Returns nan if
+        the url is invalid or null.
+        Common top level domains were pulled from this list:
+        https://www.hayksaakian.com/most-popular-tlds/
+    Examples:
+        >>> url_to_tld = URLToTLD()
+        >>> urls = ['https://www.google.com', 'http://www.google.co.in',
+        ...         'www.facebook.com']
+        >>> url_to_tld(urls).to_list()
+        ['com', 'co.in', 'com']
+    """
+    name = "url_to_tld"
+    input_types = [URL]
+    return_type = Categorical
+    filename = 'common_tlds.txt'
+
+    def get_function(self):
+        with open(self.get_filepath(self.filename), 'r') as f:
+            tlds = f.read().splitlines()
+        # put longer TLDs first to avoid catching a small part of a longer TLD
+        tlds = sorted(tlds, key=len, reverse=True)
+        # escape periods
+        tlds = [re.escape(t) for t in tlds]
+        # build regex pattern
+        self.tlds_pattern = r'(?:\.({}))'.format('|'.join(tlds))
+
+        def url_to_domain(x):
+            p = r'^(?:https?:\/\/)?(?:[^@\/\n]+@)?(?:www\.)?([^:\/?\n]+)'
+            return x.str.extract(p, expand=False)
+
+        def url_to_tld(x):
+            domains = url_to_domain(x)
+            df = domains.str.extractall(self.tlds_pattern)
+            matches = df.groupby(level=0).last()[0]
+            return matches.reindex(x.index)
+        return url_to_tld
+
+class IsFreeEmailDomain(TransformPrimitive):
+    """Determines if an email address is from a free email domain.
+    Description:
+        EmailAddress input should be a string. Will return Nan
+        if an invalid email address is provided, or if the input is
+        not a string. The list of free email domains used in this primitive
+        was obtained from https://gist.github.com/tbrianjones/5992856.
+    Examples:
+        >>> is_free_email_domain = IsFreeEmailDomain()
+        >>> is_free_email_domain(['name@gmail.com', 'name@featuretools.com']).tolist()
+        [True, False]
+    """
+    name = "is_free_email_domain"
+    input_types = [EmailAddress]
+    return_type = Boolean
+
+    filename = "free_email_provider_domains.txt"
```

Comment:
This file should be present here
https://github.com/alteryx/featuretools/tree/main/featuretools/primitives/data